src/cpp_generator.cpp

```diff
@@ -208,8 +208,24 @@ cpp_generator::function cpp_generator::generate_module(const module& m,
     f.set_name(name).set_types(m).set_body(
         m, [&](instruction_ref ins, const auto& names) -> std::string {
             if(ins->name() == "@literal")
-                return shape::cpp_type(ins->get_shape().type()) + "(" +
-                       ins->get_literal().to_string() + ")";
+            {
+                std::string string_literal;
+                ins->get_literal().visit([&](auto v) {
+                    assert(v.size() == 1);
+                    auto x = v.front();
+                    if(std::isinf(x))
+                    {
+                        string_literal = "__builtin_huge_valf()";
```

Comment:
would this work for `double` ? `valf` is `float` type.

Reply:
Yea it should probably be `__builtin_huge_val()`, and it will implicitly convert to the smaller float size.

Reply:
Done